src/Http/Routing/src/Matching/DefaultEndpointSelector.cs

```diff
@@ -60,6 +60,8 @@ private static void ProcessFinalCandidates(
         Endpoint? endpoint = null;
         RouteValueDictionary? values = null;
         int? foundScore = null;
+        var candidatesWithSameScore = new List<CandidateState>();
```

Comment:
Can we initialize this the first time two candidates with the same score are detected?